qemu/tests/block_hotplug.py

```diff
@@ -86,7 +86,14 @@ def get_new_disk(disk1, disk2):
             if params.get("need_plug") == "yes":
                 if params.get("need_controller", "no") == "yes":
                     controller_model = params.get("controller_model")
-                    controller = qdevices.QDevice(controller_model)
+                    if params.get("bus_extra_params_%s" % img_list[num + 1]):
+                        bus_extra_param = params["bus_extra_params_%s" % img_list[num + 1]]
+                        match = re.search("iothread=(\w+)", bus_extra_param)
+                        if match:
+                            qdevice_params = {"iothread": match.group(1)}
+                            controller = qdevices.QDevice(controller_model, params=qdevice_params)
+                    else:
+                        controller = qdevices.QDevice(controller_model)
```

Comment:
Better to keep line 96 in line 89 as a default value of controller.
If line 92 match is 0, the execution of line 97 could be failed.